Public/Src/Utilities/Utilities/CommandLineEscaping.cs

```diff
@@ -235,16 +235,29 @@ private static int Repeat(StringBuilder builder, string s, int count)
 
         private static bool ContainsDelimiter(string s)
         {
-            foreach (char c in s)
+            if (OperatingSystemHelper.IsUnixOS)

```

Comment:
>atingSystemHelper.IsUnixOS) [](start = 20, length = 27)

Okay for one, but if we hit more of these we should add a UnixOsCommandLine encoding scheme. This is in `\Public\Src\Pips\Dll\Operations\PipDataFragmentEscaping.cs`